### PR TITLE
Ft6x06 touch status fix and impoved report efficency

### PR DESCRIPTION
**Pull request by alexandruradovici**
### Pull Request Overview

This pull request fixes a bug in ft6x06 when reporting the touch status. It also reports touches more efficiently to the user land, scheduling a callback only when the position or status changes.

### Testing Strategy

This pull request was tested using an stm32f412g discovery.

### TODO or Help Wanted

N/A

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: capsules/src/touch.rs [hunk 1]
```diff
@@ -41,6 +43,9 @@ impl Default for App {
             events_buffer: None,
             ack: true,
             dropped_events: 0,
+            x: 0,
+            y: 0,
+            status: 3,
```
Comment (hudson-ayers) [inline capsules/src/touch.rs:48]: Could status be changed to be a `TouchStatus` type, and an `Unstarted` variant added to `TouchStatus`? I find this initialization to `3` confusing and prone to breakage in the future if additional variants are added to `TouchStatus`.

Comment (alexandruradovici) [reply capsules/src/touch.rs]: Done, thank you for the tip.

**Comment by alexandruradovici** [comment]
Hey, any updates on merging this?

**Pull request merged**